tardis/montecarlo/montecarlo_numba/interaction.py

```diff
@@ -16,7 +17,117 @@
     InteractionType,
 )
 from tardis.montecarlo.montecarlo_numba.utils import get_random_mu
-from tardis.montecarlo.montecarlo_numba.macro_atom import macro_atom
+from tardis.montecarlo.montecarlo_numba.macro_atom import (
+        macro_atom, MacroAtomTransitionType
+)
+
+
+
+@njit(**njit_dict_no_parallel)
+def scatter(r_packet, time_explosion):
+
+    old_doppler_factor = get_doppler_factor(
+        r_packet.r, r_packet.mu, time_explosion
+    )
+    comov_nu = r_packet.nu * old_doppler_factor
+    comov_energy = r_packet.energy * old_doppler_factor
+    r_packet.mu = get_random_mu()
+    inverse_new_doppler_factor = get_inverse_doppler_factor(
+        r_packet.r, r_packet.mu, time_explosion
+    )
+
+    r_packet.energy = comov_energy * inverse_new_doppler_factor
+ 
+    return comov_nu, inverse_new_doppler_factor
+
+# Maybe make the continuum selection a method of the continuum?
+@njit(**njit_dict_no_parallel)
+def continuum_event(r_packet, time_explosion, continuum, numba_plasma):
+
+    comov_nu, inverse_new_doppler_factor = scatter(r_packet, time_explosion)
+
+    old_doppler_factor = get_doppler_factor(r_packet.r, r_packet.mu, time_explosion)
+    comov_nu = r_packet.nu * old_doppler_factor
+
+    zrand = np.random.random()
+
+    # Does this need to be re-calculated?
+    # continuum.calculate(comov_nu, r_packet.current_shell_id)
+
+    # Need to determine if a collisional process or not.  If not:
+
+    # This somehow determines the transition type and continuum id needed
+    # but does not sample new frequencies
+    # This reruns continuum.calculate
+    destination_level_idx = continuum.determine_macro_activation_idx(
+            comov_nu, r_packet.current_shell_id)
+
+    transition_id, transition_type = macro_atom(
+            destination_level_idx, 
+            r_packet.current_shell_id, 
+            numba_plasma
+            )
+
+    # Then use this to get a transition id from the macroatom 
+    if transition_type == MacroAtomTransitionType.FF_EMISSION: 
+        free_free_emission(r_packet, time_explosion, numba_plasma, continuum)
+    elif transition_type == MacroAtomTransitionType.BF_EMISSION:
+        bound_free_emission(r_packet, 
+                time_explosion, 
+                numba_plasma, 
+                continuum, 
+                transition_id)
+
```

Comment:
We need to add adiabatic cooling here and probably an additional "else clause" to catch unsupported or partially supported interaction types (e.g. two-photon emission).